### DISCUSSION TITLE
Search component

### DISCUSSION THREAD

**Discussion by Nikhilchandra-Rathod** [mantinedev/mantine#2989] [Q&A]
Hey,

Can I have an example for select search functionality with ajax response.

For example: customer selector

**Comment by rtivital** [comment]
We do not have such example. Load data from the api and provide it to the `data` prop. You can subscribe to search query change with `onSearchChange` prop.

**Reply by Nikhilchandra-Rathod** [reply]
okay, I created sample here
https://codesandbox.io/s/peaceful-yonath-7fudq7?file=/src/App.tsx
Now all working except key down event not scrolling dropdown automatically.

For testing:
Type: 'te' in autocomplete and it will show many result now press keydown to select any item but it will not scroll dropdown for last items.

**Comment by baba43** [comment]
I am also trying to implement a search (in my case for an address). 

The problem with the suggested approach by @rtivital is that the results might not exacly match the value returned by the api:

If I type `"[Company name], [City]"` and search for it, the results might be formatted differently (like `"[Company], [Street], [City]"`), causing the options to be missing. What I would need in this case is the ability to provide a custom matching function to tell the AutoComplete what options are actually matching the search input. 

Is there any proper way to implement this right now?

**Reply by rtivital** [reply]
You can provide custom filter function in Autocomplete component

**Reply by baba43** [reply]
Wow, great. My bad.
